seatunnel-connectors-v2/connector-cdc/connector-cdc-base/src/main/java/org/apache/seatunnel/connectors/cdc/base/utils/CatalogTableUtils.java

```diff
@@ -0,0 +1,145 @@
+
+import org.apache.seatunnel.api.table.catalog.CatalogTable;
+import org.apache.seatunnel.api.table.catalog.Column;
+import org.apache.seatunnel.api.table.catalog.PhysicalColumn;
+import org.apache.seatunnel.api.table.catalog.PrimaryKey;
+import org.apache.seatunnel.api.table.catalog.TablePath;
+import org.apache.seatunnel.api.table.catalog.TableSchema;
+import org.apache.seatunnel.connectors.cdc.base.config.JdbcSourceTableConfig;
+
+import io.debezium.relational.Table;
+import io.debezium.relational.TableId;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class CatalogTableUtils {
+
+    public static List<CatalogTable> mergeCatalogTableConfig(
+            List<CatalogTable> tables,
+            List<JdbcSourceTableConfig> tableConfigs,
+            Function<String, TablePath> parser) {
+        Map<TablePath, CatalogTable> catalogTableMap =
+                tables.stream()
+                        .collect(Collectors.toMap(t -> t.getTableId().toTablePath(), t -> t));
+        for (JdbcSourceTableConfig catalogTableConfig : tableConfigs) {
+            TablePath tablePath = parser.apply(catalogTableConfig.getTable());
+            CatalogTable catalogTable = catalogTableMap.get(tablePath);
+            if (catalogTable != null) {
+                catalogTable = mergeCatalogTableConfig(catalogTable, catalogTableConfig);
+                catalogTableMap.put(tablePath, catalogTable);
+                log.info(
+                        "Override primary key({}) for catalog table {}",
+                        catalogTableConfig.getPrimaryKeys(),
+                        catalogTableConfig.getTable());
+            } else {
+                log.warn(
+                        "Table {} is not found in catalog tables, skip to merge config",
+                        catalogTableConfig.getTable());
+            }
+        }
+        return new ArrayList<>(catalogTableMap.values());
+    }
+
+    public static CatalogTable mergeCatalogTableConfig(
+            final CatalogTable table, JdbcSourceTableConfig config) {
+        List<String> columnNames =
+                table.getTableSchema().getColumns().stream()
+                        .map(c -> c.getName())
+                        .collect(Collectors.toList());
+        for (String pk : config.getPrimaryKeys()) {
+            if (!columnNames.contains(pk)) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Primary key(%s) is not in table(%s) columns(%s)",
+                                pk, table.getTablePath(), columnNames));
+            }
+        }
+        PrimaryKey primaryKeys =
+                PrimaryKey.of(
+                        "pk" + Math.abs(config.getPrimaryKeys().hashCode()),
```

Comment:
```suggestion
                        "pk" + (config.getPrimaryKeys().hashCode() & Integer.MAX_VALUE),
```
When hashcode values is Integer.MIN_VALUE overflow occurs, it may be better to use hashcode & Integer.MAX_VALUE